LXStudio-IDE/src/main/java/titanicsend/pattern/tom/Pulse.java

```diff
@@ -0,0 +1,119 @@
+package titanicsend.pattern.tom;
+
+import heronarts.lx.LX;
+import heronarts.lx.color.LXColor;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.modulator.SawLFO;
+import heronarts.lx.parameter.*;
+import heronarts.lx.transform.LXVector;
+import titanicsend.model.TEPanelModel;
+import titanicsend.model.TEVertex;
+import titanicsend.pattern.TEPattern;
+import titanicsend.util.PanelStriper;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Pulse extends TEPattern {
+    private HashMap<String, LXPoint[][]> pointMap;
+
+    protected final CompoundParameter rate = (CompoundParameter)
+            new CompoundParameter("Rate", .25, .01, 2)
+                    .setExponent(2)
+                    .setUnits(LXParameter.Units.HERTZ)
+                    .setDescription("Rate of the rotation");
+
+    protected final SawLFO phase = new SawLFO(0, 1, new FunctionalParameter() {
+        public double getValue() {
+            return 1000 / rate.getValue();
+        }
+    });
+
+    public Pulse(LX lx) {
+        super(lx);
+        startModulator(this.phase);
+        addParameter("rate", this.rate);
+        pointMap = buildPointMap(model.panelsById);
+    }
+
+    public void run(double deltaMs) {
+        float phase = this.phase.getValuef();
+        for (Map.Entry<String, TEPanelModel> entry : model.panelsById.entrySet()) {
+            LXPoint[][] panelPoints = pointMap.get(entry.getKey());
+            int litIndex = (int) (phase * (panelPoints.length - 1));
+            LXPoint[] litSection = panelPoints[litIndex];
+
+            for (int i = 0; i < entry.getValue().points.length; i++) {
+                colors[entry.getValue().points[i].index] = LXColor.BLACK;
+            }
+
+            for (LXPoint point : litSection) {
+                colors[point.index] = LXColor.WHITE;
+            }
+        }
+    }
+
+    private HashMap<String, LXPoint[][]> buildPointMap(HashMap<String, TEPanelModel> panels) {
+        return panels.entrySet().stream()
+                .collect(Collectors.toMap(
+                        e -> e.getKey(),
+                        e -> buildPanelMap(e.getValue()),
+                        (a, b) -> a,
+                        HashMap::new
+                ));
+    }
+
+
+    private LXPoint[][] buildPanelMap(TEPanelModel panel) {
+        ArrayList<ArrayList<LXPoint>> points = new ArrayList<ArrayList<LXPoint>>();
+
+        TEVertex[] currentVertices = {
+                new TEVertex(new LXVector(panel.v0), -1),
+                new TEVertex(new LXVector(panel.v1), -1),
+                new TEVertex(new LXVector(panel.v2), -1)
+        };
+
+        int i = 1;
+        while (currentVertices[0].distanceTo(panel.centroid) > (2 * PanelStriper.DISTANCE_BETWEEN_PIXELS)) {
```

Comment:
Note that the centroid is not necessarily equidistant to all three vertexes. Does that matter to the pattern?

Also we should agree on whether to call them vertexes or vertices. Or just call them Nodes and avoid the debate. :D

Reply:
It seems to work alright for all the panels, I think specifically because we're moving the vertexes by a percentage of their distance from the centroid at a time rather than by the same absolute distance for each vertex.

'vertexes' seems to be occur more frequently than 'vertices' in the repo currently, so I'm cool with that.